internal/services/managedapplications/managed_application_definition_resource.go

```diff
@@ -193,105 +190,99 @@ func resourceManagedApplicationDefinitionRead(d *pluginsdk.ResourceData, meta in
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.ApplicationDefinitionID(d.Id())
+	id, err := applicationdefinitions.ParseApplicationDefinitionID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	resp, err := client.Get(ctx, *id)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			log.Printf("[INFO] Managed Application Definition %q does not exist - removing from state", d.Id())
 			d.SetId("")
 			return nil
 		}
-		return fmt.Errorf("failed to read Managed Application Definition %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+		return fmt.Errorf("failed to read Managed Application Definition %s: %+v", id, err)
```

Comment:
_minor_ but for consistency:

```suggestion
		return fmt.Errorf("retrieving %s: %+v", id, err)
```